### ISSUE TITLE
ALB-079: CUDA trainer ignores lr_scheduler — constant lr after warmup

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#241]
## Bug

`CudaTransformerTrainer::current_lr()` only implements linear warmup. After warmup completes, it returns `base_lr` (constant) regardless of the `lr_scheduler: "cosine"` YAML config.

## Five Whys

1. **Why val_loss plateau at 6.92?** Model stuck in local minimum, gnorm collapsed 3.0→0.13
2. **Why gnorm collapse?** Loss landscape is flat at constant lr=3e-4 — model can't escape
3. **Why constant lr?** `current_lr()` only has warmup logic (lines 1934-1941), no cosine decay
4. **Why no cosine?** `TransformerTrainConfig` has no `lr_scheduler` field. YAML config parsed by bridge but never passed to CUDA trainer
5. **Why not caught?** The log shows `lr=3.00e-4` at all steps — but early in training (step 2K-5K) cosine decay is nearly invisible (lr≈2.99e-4)

## Evidence

```
step  1000: lr=1.50e-4 (warmup, correct)
step  2000: lr=3.00e-4 (end of warmup, correct)
step 26000: lr=3.00e-4 ← SHOULD BE ~2.93e-4 (cosine)
step 125K:  lr=3.00e-4 ← SHOULD BE ~1.50e-4 (cosine midpoint)
step 250K:  lr=3.00e-4 ← SHOULD BE ~0 (cosine end)
```

Val loss plateau at 6.92 since step 12K strongly correlates with constant lr.

## Fix

Add cosine decay to `CudaTransformerTrainer::current_lr()` using warmup_steps and max_steps.

```rust
pub fn current_lr(&self) -> f32 {
    let base_lr = self.config.lr;
    if self.step < self.config.warmup_steps {
        base_lr * (self.step as f32 / self.config.warmup_steps.max(1) as f32)
    } else {
        let max_steps = self.config.max_steps.unwrap_or(self.step + 1);
        let decay_steps = max_steps.saturating_sub(self.config.warmup_steps);
        let decay_step = self.step - self.config.warmup_steps;
        let progress = (decay_step as f32 / decay_steps.max(1) as f32).min(1.0);
        let cosine_decay = 0.5 * (1.0 + (std::f32::consts::PI * progress).cos());
        base_lr * cosine_decay
    }
}
```

## Impact

- Current v3 training (26K/250K steps, 10.4% complete) is at constant lr
- Fix requires training restart — sunk 26K steps still provide good checkpoint
- With proper cosine decay, expect val_ppl to break through 1000 floor
- No change to checkpoint format — can resume from step-26K checkpoint after fix
